include/flox/core/exceptions.hh

```diff
@@ -74,76 +74,150 @@ enum error_category {
+  class NAME : public FloxException                                            \
+  {                                                                            \
+  public:                                                                      \
+                                                                               \
+    NAME() : FloxException( CATEGORY_MSG, std::nullopt, std::nullopt )         \
+    {}                                                                         \
+                                                                               \
+    explicit NAME( std::string_view contextMsg )                               \
+      : FloxException( CATEGORY_MSG, std::string( contextMsg ), std::nullopt ) \
+    {}                                                                         \
+                                                                               \
+    explicit NAME( std::string_view contextMsg, std::string_view caughtMsg )   \
+      : FloxException( CATEGORY_MSG,                                           \
+                       std::string( contextMsg ),                              \
+                       std::string( caughtMsg ) )                              \
+    {}                                                                         \
+                                                                               \
+    [[nodiscard]] error_category                                               \
+    getErrorCode() const noexcept override                                     \
+    {                                                                          \
+      return ERROR_CODE;                                                       \
+    }                                                                          \
+                                                                               \
+    [[nodiscard]] std::string_view                                             \
+    getCategoryMessage() const noexcept override                               \
+    {                                                                          \
+      return CATEGORY_MSG;                                                     \
+    }                                                                          \
+                                                                               \
+  };
+
+
 /* -------------------------------------------------------------------------- */
 
 /** @brief A `nix::EvalError` was encountered.  */
 class NixEvalException : public FloxException
 {
 
-private:
-
-  static constexpr std::string_view categoryMsg = "invalid argument";
-
 public:
 
   explicit NixEvalException( std::string_view      contextMsg,
                              const nix::EvalError &err )
-    : FloxException( contextMsg,
-                     nix::filterANSIEscapes( err.what(), true ).c_str() )
+    : FloxException( "invalid argument",
```

Comment:
This was (my 😳 ) copy paste error but it should be "Nix evaluation error" or something